### PR TITLE
Update IAR support version to 8.32

### PR DESCRIPTION
**Pull request by deepikabhavnani**
DO NOT MERGE - UPCOMING RELEASE

### CODE REVIEW

File: docs/tools/tools_intro.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ Mbed OS 5 can be built with various toolchains. The currently supported versions
 
 - [Arm compiler 5.06 update 6](https://developer.arm.com/products/software-development-tools/compilers/arm-compiler-5/downloads).
 - [GNU Arm Embedded version 6](https://developer.arm.com/open-source/gnu-toolchain/gnu-rm/downloads).
```
Comment (JanneKiiskila) [inline docs/tools/tools_intro.md]: Please specify which version of GNU Arm toolchain as well, there's multiple ones and the page seems to propose Version 8-2018-q4-major. Which is or is not aligned with anything else?

**Review by JanneKiiskila — CHANGES_REQUESTED** [review]

**Comment by JanneKiiskila** [comment]
If you'all would want an easy way with the 5.12 docs, you would make a 5.12 branch and point all PRs against that (which would be relevant for 5.12, that is).

File: docs/tools/tools_intro.md (hunk 1 above)
Comment (cmonr) [reply docs/tools/tools_intro.md]: @AnotherButler If it helps, we'll be specifically using `GCC Arm Compiler 6-2017-q2-update` (https://github.com/ARMmbed/mbed-os/pull/9844/files#diff-354f30a63fb0907d4ad57269548329e3R30)

https://developer.arm.com/open-source/gnu-toolchain/gnu-rm/downloads

The GNU Arm Embedded is what the Ubuntu package calls the compiler. I don't know why they name them differently.

Comment (cmonr) [reply docs/tools/tools_intro.md]: @ARMmbed/mbed-os-test-team Can confirm, but I suspect this is the same version in Jenkins CI as well.

Comment (deepikabhavnani) [reply docs/tools/tools_intro.md]: GCC version in CI is 6-2017-q1-update

**Comment by deepikabhavnani** [comment]
@AnotherButler - Please review. Not dependent on code anymore. feature-iar8 branch is merged to master

File: docs/tools/toolchains/export_to_third_party.md [hunk 2]
```diff
@@ -5,11 +5,11 @@ If you'd like to develop on Arm Mbed OS with a third party tool, or migrate to o
 - Keil uVision5.
 - IAR Systems.
 - Make using:
-    - GNU Arm Embedded Toolchain.
+    - GNU Arm Embedded Toolchain 6 (6-2017-q1-update).
     - Arm Compiler 5.
```
Comment (JanneKiiskila) [inline docs/tools/toolchains/export_to_third_party.md]: It's not just 5, it's 5.06 update 6.

File: docs/tools/CLI/cli-setup/cli-reqs.md [hunk 3]
```diff
@@ -225,7 +225,7 @@ Mbed CLI supports a setting for each toolchain path:
 | --------- | --------- | ---------| --------- |
 | Arm Compiler 5 | `C:/Program Files/ARM_Compiler_5.06u5/bin/armcc` | `ARM_PATH` | `C:/Program Files/ARM_Compiler_5.06u5` |
```
Comment (deepikabhavnani) [inline docs/tools/CLI/cli-setup/cli-reqs.md]: @cmonr @timurh01  - ARM Compiler 5 supported and in CI is ARM_Compiler_5.06u5 or ARM_Compiler_5.06u6

File: docs/tools/toolchains/export_to_third_party.md (hunk 2 above)
Comment (deepikabhavnani) [reply docs/tools/toolchains/export_to_third_party.md]: @cmonr @timurh01 - ARM Compiler 5 supported and in CI is ARM_Compiler_5.06u5 or ARM_Compiler_5.06u6

Comment (deepikabhavnani) [reply docs/tools/toolchains/export_to_third_party.md]: > It's not just 5, it's 5.06 update 6.

Done

**Review by AnotherButler — APPROVED** [review]
LGTM 👍

**Comment by AnotherButler** [comment]
@cmonr  @JanneKiiskila Have your concerns been addressed?

**Comment by AnotherButler** [comment]
Also, we make the 5.12 branch out of the development branch, but we don't do this until code freeze.

**Review by JanneKiiskila — APPROVED** [review]

**Review by cmonr — APPROVED** [review]

**Pull request merged**